File: intime/app/src/main/java/com/vpe_soft/intime/intime/view/CardViewStateHelper.java
package com.vpe_soft.intime.intime.view;

import android.view.ViewOutlineProvider;

import com.vpe_soft.intime.intime.activity.Dimensions;
import com.vpe_soft.intime.intime.recyclerview.TaskRecyclerViewAdapter;

public class CardViewStateHelper {

    private ViewOutlineProvider defaultProvider;

    private boolean isIdle = true;

    public void setDefaultProvider(ViewOutlineProvider defaultProvider) {
        this.defaultProvider = defaultProvider;
    }

    public void setDefaultState(TaskRecyclerViewAdapter.TaskRecyclerViewVH holder) {
        if (!isIdle) {
            isIdle = true;
            holder.card.setRadius(0);
            holder.card.setOutlineProvider(null);
        }
    }

    public void setOnSwipeState(TaskRecyclerViewAdapter.TaskRecyclerViewVH holder) {
        if (isIdle) {
            isIdle = false;
            holder.card.setRadius(Dimensions.cardCornerRadius);
            holder.card.setOutlineProvider(defaultProvider);
        }
    }
}

File: intime/app/src/main/java/com/vpe_soft/intime/intime/view/CardViewStateHelper.kt
package com.vpe_soft.intime.intime.view

import android.view.ViewOutlineProvider
import com.vpe_soft.intime.intime.activity.cardCornerRadius
import com.vpe_soft.intime.intime.recyclerview.TaskRecyclerViewAdapter.TaskRecyclerViewVH

//todo: rewrite this
class CardViewStateHelper {
    var defaultProvider: ViewOutlineProvider? = null
    private var isIdle = true

    fun setDefaultState(holder: TaskRecyclerViewVH) {
        if (!isIdle) {
            isIdle = true
            with(holder.card) {
                radius = 0f
                outlineProvider = null
            }
        }
    }

    fun setOnSwipeState(holder: TaskRecyclerViewVH) {
        if (isIdle) {
            isIdle = false
            with(holder.card) {
                radius = cardCornerRadius
                outlineProvider = defaultProvider
            }
        }
    }
}